### PR TITLE
Add pull requests

### PR DESCRIPTION
**Pull request by adamjonas**
ref #46 

Issues don't have merge information on them, which means that PRs need to be pulled separately.

Did my best to mimic the API of issues.

### CODE REVIEW

**Comment by adamjonas** [comment]
ref #46

**Comment by simonw** [comment]
Sorry for not looking at this sooner, trying it out now - pull request looks great!

File: tests/pull_requests.json [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+            "href": "https://api.github.com/repos/simonw/datasette/statuses/a85239f69261c10f1a9f90514c8b5d113cb94585"
+          }
+        },
+        "author_association": "OWNER",
+        "active_lock_reason": null,
+        "merged": true,
+        "mergeable": null,
+        "rebaseable": null,
+        "mergeable_state": "unknown",
+        "merged_by": {
```
Comment (simonw) [inline tests/pull_requests.json:342]: Running this should create a `merged_by` column on the `pull_requests` table which is a foreign key to the `users` table.

File: tests/test_pull_requests.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            'base': 'f04deebec4f3842f7bd610cd5859de529f77d50e',
+            'author_association': 'OWNER',
+            'merged': 1,
+            'mergeable': None,
+            'rebaseable': None,
+            'mergeable_state': 'unknown',
+            'merged_by': '{"login": "simonw", "id": 9599, "node_id": "MDQ6VXNlcjk1OTk=", "avatar_url": "https://avatars0.githubusercontent.com/u/9599?v=4", "gravatar_id": "", "url": "https://api.github.com/users/simonw", "html_url": "https://github.com/simonw", "followers_url": "https://api.github.com/users/simonw/followers", "following_url": "https://api.github.com/users/simonw/following{/other_user}", "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}", "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/simonw/subscriptions", "organizations_url": "https://api.github.com/users/simonw/orgs", "repos_url": "https://api.github.com/users/simonw/repos", "events_url": "https://api.github.com/users/simonw/events{/privacy}", "received_events_url": "https://api.github.com/users/simonw/received_events", "type": "User", "site_admin": false}',
```
Comment (simonw) [inline tests/test_pull_requests.py:76]: See above - this should be 9599, an integer reference to the row in the users table.

File: tests/test_pull_requests.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            table="pull_requests", column="user", other_table="users", other_column="id"
+        ),
+    } == set(db["pull_requests"].foreign_keys)
+
+
+def test_pull_requests(db):
+    pull_request_rows = list(db["pull_requests"].rows)
+    assert [
+        {
+            'id': 313384926,
```
Comment (simonw) [inline tests/test_pull_requests.py:53]: Minor detail: I use Black for this repo, which requires double quotes - running "black ." in the root directory (with the latest version of Black) should handle this for you.

**Review by simonw — CHANGES_REQUESTED** [review]

File: tests/pull_requests.json (hunk 1 above)
Comment (adamjonas) [reply tests/pull_requests.json:342]: Nice eye. This is a little misleading. The merged by field is only available for specified PRs (singular), which is how I generated this. There are a different set of keys when requesting PRs (plural) via the API. Happy to modify the payload here, but merged_by is not included. For reference, [here is a gist](https://gist.github.com/adamjonas/5ee0ce97f091e8e36931d5e8d7eb6502) of what the first PR from the pull-requests request looks like.

File: tests/test_pull_requests.py (hunk 3 above)
Comment (adamjonas) [reply tests/test_pull_requests.py]: ran black as suggested

**Pull request merged**